worlds/metroidprime/data/PhazonMines.py

```diff
@@ -0,0 +1,823 @@
+                        RoomName.Security_Access_B,
+                        defaultLock=DoorLockType.Wave,
+                        rule_func=can_wave_beam,
+                    ),
+                    2: DoorData(
+                        RoomName.Storage_Depot_A,
+                        defaultLock=DoorLockType.Plasma,
+                        rule_func=lambda world, state: can_power_bomb(world, state)
+                        and can_plasma_beam(world, state)
+                        and can_scan(world, state),
+                    ),
+                }
+            ),
+            RoomName.Missile_Station_Mines: RoomData(
+                doors={
+                    0: DoorData(
+                        RoomName.Fungal_Hall_B, defaultLock=DoorLockType.Plasma
+                    ),
+                }
+            ),
+            RoomName.Omega_Research: RoomData(
+                doors={
+                    0: DoorData(
+                        RoomName.Map_Station_Mines,
+                        defaultLock=DoorLockType.Ice,
+                        rule_func=lambda world, state: can_combat_mines(world, state)
+                        and can_power_bomb(world, state),
+                    ),
+                    1: DoorData(
+                        RoomName.Ventilation_Shaft,
+                        defaultLock=DoorLockType.Ice,
+                        rule_func=lambda world, state: can_combat_mines(world, state)
+                        and can_power_bomb(world, state),
+                    ),
+                    2: DoorData(
+                        RoomName.Dynamo_Access,
+                        defaultLock=DoorLockType.Ice,
+                        destination_area=MetroidPrimeArea.Phazon_Mines,
+                        rule_func=lambda world, state: can_combat_mines(world, state)
+                        and can_power_bomb(world, state),
+                    ),  # Vertical door going down
+                }
+            ),
+            RoomName.Ore_Processing: RoomData(
+                doors={
+                    0: DoorData(
+                        RoomName.Research_Access,
+                        defaultLock=DoorLockType.Ice,
+                        rule_func=lambda world, state: True,
+                    ),
+                    1: DoorData(
+                        RoomName.Storage_Depot_B,
+                        defaultLock=DoorLockType.Ice,
+                        rule_func=lambda world, state: (
+                            can_combat_beam_pirates(
+                                world, state, SuitUpgrade.Power_Beam
+                            )
+                            and can_spider(world, state)
+                            and can_bomb(world, state)
+                            and can_power_bomb(world, state)
+                        )
+                        or (
+                            state.can_reach(
+                                RoomName.Waste_Disposal.value, None, world.player
```

Comment:
This access rule is being used on entrances, e.g. `Ice Beam Door from Ore Processing to Storage Depot B`, in which case, the `"Waste Disposal"` region must be registered as an indirect condition for the entrance, or the world has to use `explicit_indirect_conditions = False` and take a performance hit to its reachable region updates.

Alternatively, if adding the indirect condition is complicated, an event item/location pair could be placed within the `"Waste Disposal"` region and then the access rule can instead check for state having collected that event item. Note that events are usually worse performance due to introducing an extra location, and the event can create an extra logical sphere because the item must be collected within a sphere to move onto the next sphere where the entrance becomes accessible.